### PR TITLE
Add latest Rubocop rules

### LINKED ISSUE #14465: The following cops were added to RuboCop, but are not configured
```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that can also opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable

Lint/DuplicateBranch: # (new in 1.3)
  Enabled: true
Lint/DuplicateRegexpCharacterClassElement: # (new in 1.1)
  Enabled: true
Lint/EmptyBlock: # (new in 1.1)
  Enabled: true
Lint/EmptyClass: # (new in 1.3)
  Enabled: true
Lint/NoReturnInBeginEndBlocks: # (new in 1.2)
  Enabled: true
Lint/ToEnumArguments: # (new in 1.1)
  Enabled: true
Lint/UnexpectedBlockArity: # (new in 1.5)
  Enabled: true
Lint/UnmodifiedReduceAccumulator: # (new in 1.1)
  Enabled: true
Style/ArgumentsForwarding: # (new in 1.1)
  Enabled: true
Style/CollectionCompact: # (new in 1.2)
  Enabled: true
Style/DocumentDynamicEvalDefinition: # (new in 1.1)
  Enabled: true
Style/NegatedIfElseCondition: # (new in 1.2)
  Enabled: true
Style/NilLambda: # (new in 1.3)
  Enabled: true
Style/RedundantArgument: # (new in 1.4)
  Enabled: true
Style/SwapValues: # (new in 1.1)
  Enabled: true
For more information: https://docs.rubocop.org/rubocop/versioning.html
```

### PR DESCRIPTION
**Pull request by adfoster-r7**
Continuation of https://github.com/rapid7/metasploit-framework/pull/14639
Resolves #14465

### CODE REVIEW

File: modules/auxiliary/admin/http/ulterius_file_download.rb [hunk 1]
```diff
@@ -52,6 +52,30 @@ def process_data(index, parse_data)
...
+    def zlib(level=1, window_bits=2, mem_level=3, strategy=4)
+        puts "level #{level}"
+        puts "window_bits #{window_bits}"
+        puts "mem_level #{mem_level}"
+        puts "strategy #{strategy}"
+    end
+
+    zlib(window_bits =-15)
+=end
     zi = Zlib::Inflate.new(window_bits =-15)
```
Comment (adfoster-r7) [inline modules/auxiliary/admin/http/ulterius_file_download.rb:79]: Will await a second pair of eyes on this, this is currently:
```
zi = Zlib::Inflate.new(-15)
```

Which is probably unintended, but it worked at the time of module landing I imagine.

Comment (bcoles) [reply modules/auxiliary/admin/http/ulterius_file_download.rb:79]: > 
> 
> Will await a second pair of eyes on this, this is currently:
> 
> ```
> zi = Zlib::Inflate.new(-15)
> ```
> 
> Which is probably unintended, but it worked at the time of module landing I imagine.

This doesn't look right.

I have no idea what the intention  of the original code was either.

```
$ cat test.rb 
#!/usr/bin/env ruby

def test(a = 1, b = 2, c = 3, d = 4)
  puts "a: #{a}"
  puts "b: #{b}"
  puts "c: #{c}"
  puts "d: #{d}"
end


test(b = 1234)
$ ./test.rb 
a: 1234
b: 2
c: 3
d: 4
```

Comment (adfoster-r7) [reply modules/auxiliary/admin/http/ulterius_file_download.rb]: Looks like I accidentally grabbed the method definition for the `Deflate` method, the inflate one is much easier to work out:

```
        def initialize(window_bits = Zlib::MAX_WBITS)
            # ...
        end
```

Fixed now :+1:

**Comment by dwelch-r7** [comment]
# Release Notes

Adds the latest rubocop rules

**Pull request merged**